src/main.ts

```diff
@@ -54,13 +59,55 @@ async function initializeVue() {
     store.dispatch('fetchPolicies'),
   ])
 
-  const router = await createRouter(store.state.policies)
+  const guiBasePath = readGuiBasePathFromDom()
```

Comment:
How come we need this (and the function itself) I understood we can get almost the same value from `pathConfig.baseGuiPath` just without the trailing slash, which we can just add on the end? In fact I think there's something a little of here considering the current default fallback in the `readGuiBasePathFromDom` function doesn't end in a slash and the `<base href="">` value will always end in a slash.

Reply:
You’re right; it makes more sense to read this from the config object as well.

One thing that’s somewhat bothersome now is the following:

* With replacing the config in non-production environments in the Vite configuration, we need to define the default path config object there. Specifically, we need to define it _after the environment variables have been populated_ because the object needs to have access to `VITE_KUMA_API_SERVER_URL`.
* This means we can’t define this object in `src/constants.ts` and import it in vite.config.js because the import would happen _before_ we have a chance to run `dotenv.config()` which allows us access to environment variables in vite.config.js.
* This means we can’t import the same default config in main.ts

Reply:
> This means we can’t import the same default config in main.ts

Maybe I don't follow but my question would be, why do we need a default config in main.ts? We rely on the config always being set and passed to us from the backend (or the build tool during dev/testing/preview), so in that case there is no 'default' as the config is always passed to us. There shouldn't be a case where we aren't passed a config.

Reply:
Yes, this is ideally the case, but this is a critical point of failure: if for some reason, the templating fails, the entire application breaks immediately despite us being able to offer most users a working product seeing that the majority of people do use the quasi standard /gui GUI base path and the default API URLs. I consider this a defensive measure against this unlikely but severe risk.

Reply:
In the unlikely event of the templating failing should that not be dealt with in the backend? I'm only saying as it means the problem you've outlined above becomes a non-problem and we don't have to worry about it.

Reply:
How do you imagine the backend anticipating a failure in templating? I don’t think that makes more sense than doing it right here. There is nothing inherently wrong with duplicating this structure and while it would be nice to re-use the same object, we can’t (at least I don’t know how).

Reply:
> How do you imagine the backend anticipating a failure in templating?

Well thats up to backend I would say, but at a guess they could just fallback to a default JSON the same as you are suggesting we do here (or even a default non-templated HTML file). Honestly I don't think this is ever going to be a major issue, and if it does I'd say it should be surfaced in the backend and dealt with there i.e. fix it where it broke.

All this aside, is this is causing no issues for you or you have a workaround you are happy with then I really don't mind either way.